src/model/Model.js

```
@@ -380,7 +380,13 @@ export default class Model {
 
 /** Lambda expression for getting an instance Ref out of a match result. */
 function indexRefGetter(index) {
-  return index.values ?
-    arr => query.get(query.select(index.values.length, arr)) :
-    query.get
+  if (index.values) {
+    var idx = -1
+    for (var i = 0; i < index.values.length; i++) {
+      if (index.values[i]['path'] == 'ref') idx = i
+    }
```

If values are defined on the index, refs are no long implicitly covered. They must be covered explicitly. Should we throw an exception here if the index isn't covering refs?